solutions/USACO jan20-wormholesort.mdx

```diff
@@ -2,15 +2,115 @@
 id: usaco-992
 source: USACO Silver 2020 January
 title: Wormhole Sort
-author: Óscar Garries
+author: Óscar Garries, Neo Wang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_wormsort_silver_jan20.html)
 
+# Alternative Solution (DSU/UF)
+
+Like the DFS solution, we binary search on the answer $x$. $x$ is valid if all $p_i$ are in the same component as $i$, which we can query in $\mathcal{O}(\alpha)$ using a Union Find/Disjoint Set data structure.
+
+Our resulting time complexity is $\mathcal{O}(NlogN)$
```

Comment:
\log N?

Reply:
Note that the UF solution here runs in linear time:

https://usaco.guide/general/factors-choosing#time-limit

Reply:
(after sorting edges)